src/_pytest/config/findpaths.py

```diff
@@ -93,57 +96,57 @@ def locate_config(
     ]
     args = [x for x in args if not str(x).startswith("-")]
     if not args:
-        args = [py.path.local()]
+        args = [Path.cwd()]
     for arg in args:
-        arg = py.path.local(arg)
-        for base in arg.parts(reverse=True):
+        argpath = Path(arg).resolve()
```

Comment:
I think we should avoid `resolve()` in general because of what we learned in https://github.com/pytest-dev/pytest/pull/6523. I think this will provide the same results:

```suggestion
        argpath = Path(arg).absolute()
```

Reply:
`Path.absolute` is not public, and also doesn't resolve `.` and `..` which is necessary here. It also has XXX and FIXME in its implementation, seems like better avoided.

`py.path.local(...)` does an `os.path.abspath` internally, which does normalize.

If `resolve()` is no good and we want to keep the behavior I guess we'll need to do `Path(abspath(...))` instead.

Reply:
> If resolve() is no good

Yeah I'm afraid that's the case.

I suggest we add a `def absolute(p: Union[Path, str]) -> Path` to `_pytest.pathlib` which does `Path(abspath(...))` internally and use that then.

Having a separate function will avoid having to remember how to do it each time and also a good place to document why it exists in the first place.

Reply:
Agreed, I added `absolutepath` (see commit message in regards to the name).

Reply:
Perfect! 👍